pkg/controller/render/render_controller.go

```diff
@@ -1018,216 +770,53 @@ func (ctrl *Controller) experimentalRenderToImageStream(pool *mcfgv1.MachineConf
-// stuffs them into a machineconfig for our layered pool, so we can have entitled builds. This is a terrible practice, and
-// we should probably just sync the secrets into our namespace so our builds can use them directly rather than expose them via machineconfig.
-func (ctrl *Controller) experimentalAddEntitlements(pool *mcfgv1.MachineConfigPool) {
-
-	var entitledConfigName = fmt.Sprintf("99-%s-entitled-build", pool.Name)
-
-	// If it's not there, put it there, otherwise do nothing
-	_, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), entitledConfigName, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-
-		// Repo configuration for redhat package entitlements ( I just added base and appstream)
-		// TODO(jkyros): do this right once subscription-manager is included in RHCOS
-		redhatRepo := `[rhel-8-for-x86_64-baseos-rpms]
-name = Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)
-baseurl = https://cdn.redhat.com/content/dist/rhel8/8/x86_64/baseos/os
-enabled = 1
-gpgcheck = 0
-sslverify = 0
-sslclientkey = /etc/pki/entitlement/entitlement-key.pem
-sslclientcert = /etc/pki/entitlement/entitlement.pem
-metadata_expire = 86400
-enabled_metadata = 1
-
-[rhel-8-for-x86_64-appstream-rpms]
-name = Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)
-baseurl = https://cdn.redhat.com/content/dist/rhel8/8/x86_64/appstream/os
-enabled = 1
-gpgcheck = 0
-sslverify = 0
-sslclientkey = /etc/pki/entitlement/entitlement-key.pem
-sslclientcert = /etc/pki/entitlement/entitlement.pem
-metadata_expire = 86400
-enabled_metadata = 1
-`
-
-		// Make an ignition to stuff into our machineconfig
-		ignConfig := ctrlcommon.NewIgnConfig()
-		ignConfig.Storage.Files = append(ignConfig.Storage.Files, NewIgnFile("/etc/yum.repos.d/redhat.repo", redhatRepo))
-
-		// Get our entitlement secrets out of the managed namespace
-		entitlements, err := ctrl.kubeclient.CoreV1().Secrets("openshift-config-managed").Get(context.TODO(), "etc-pki-entitlement", metav1.GetOptions{})
-		if err != nil {
-			glog.Warningf("Could not retrieve entitlement secret: %s", err)
-			return
-		}
-
-		// Add the key to the file list
-		if key, ok := entitlements.Data["entitlement-key.pem"]; ok {
-			ignConfig.Storage.Files = append(ignConfig.Storage.Files, NewIgnFile("/etc/pki/entitlement/entitlement-key.pem", string(key)))
-		}
 
-		// Add the public key to the file list
-		if pub, ok := entitlements.Data["entitlement.pem"]; ok {
-			ignConfig.Storage.Files = append(ignConfig.Storage.Files, NewIgnFile("/etc/pki/entitlement/entitlement.pem", string(pub)))
-		}
-
-		// Now it's a machineconfig
-		mc, err := ctrlcommon.MachineConfigFromIgnConfig(pool.Name, entitledConfigName, ignConfig)
-		if err != nil {
-			glog.Warningf("Could not create machineconfig for entitlements: %s", err)
+	var mostRecent *imagev1.TagEvent
+	// Get the most recent image
+	for _, tag := range poolImageStream.Status.Tags {
+		if tag.Tag == "latest" {
```

Comment:
I'm not familiar with the Image API or how the build controller is consuming this, so not sure if this would be possible...but I'm wondering if we could use generated.Name as a tag and then also tag that as latest. Then if I delete a MC, rolling back to the previously rendered config is faster, because you could just check that the image with that tag already exists and tag it as latest.